shared/src/main/scala/VList.scala

```diff
@@ -100,6 +100,14 @@ class VList private (val lst: Seq[VAny])
   override def toString(): String =
     lst.map(_.toString).mkString("[ ", " | ", " ]")
 
+  def toStringWithContext()(using ctx: Context): String =
+    lst
+      .map(_ match
+        case n: VNum => NumberHelpers.numToString(n)
+        case x       => x.toString()
+      )
```

Comment:
```suggestion
      .map {
        case n: VNum => NumberHelpers.numToString(n)
        case x       => x.toString
      }
```
Use a pattern matching function literal if your only argument is a function that does pattern matching

Reply:
Also, this method won't handle nested `VList`s